engine/common/synchronization/finalized_snapshot.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/lifecycle"
+	"github.com/onflow/flow-go/state/protocol"
+)
+
+// finalizedSnapshot is a helper structure which contains latest finalized header and participants list
+type finalizedSnapshot struct {
+	head         *flow.Header
+	participants flow.IdentityList
+}
+
+// FinalizedSnapshotCache represents a cached snapshot of the latest finalized header and participants list.
+// It is used in Engine to access latest valid data.
+type FinalizedSnapshotCache struct {
+	mu sync.Mutex
+
+	log                       zerolog.Logger
+	state                     protocol.State
+	identityFilter            flow.IdentityFilter
+	lastFinalizedSnapshot     *finalizedSnapshot
+	finalizationEventNotifier engine.Notifier // notifier for finalization events
+
+	lm                *lifecycle.LifecycleManager
+	shutdownCompleted chan struct{} // used to signal that shutdown of finalization processing loop was completed
+
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+// NewFinalizedSnapshotCache creates a new finalized snapshot cache.
+func NewFinalizedSnapshotCache(log zerolog.Logger, state protocol.State, participantsFilter flow.IdentityFilter) (*FinalizedSnapshotCache, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	cache := &FinalizedSnapshotCache{
+		state:             state,
+		identityFilter:    participantsFilter,
+		lm:                lifecycle.NewLifecycleManager(),
+		ctx:               ctx,
+		cancel:            cancel,
+		shutdownCompleted: make(chan struct{}),
+		log:               log.With().Str("component", "finalized_snapshot_cache").Logger(),
+	}
+
+	err := cache.updateSnapshot()
+	if err != nil {
+		return nil, fmt.Errorf("could not apply last finalized state")
+	}
+
+	return cache, nil
+}
+
+// get returns last locally stored snapshot which contains final header
+// and list of filtered identities
+func (f *FinalizedSnapshotCache) get() *finalizedSnapshot {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	return f.lastFinalizedSnapshot
+}
+
+// updateSnapshot updates latest locally cached finalized snapshot
+func (f *FinalizedSnapshotCache) updateSnapshot() error {
+	finalSnapshot := f.state.Final()
+	head, err := finalSnapshot.Head()
+	if err != nil {
+		return fmt.Errorf("could not get last finalized header: %w", err)
+	}
+
+	// get all participant nodes from the state
+	participants, err := finalSnapshot.Identities(f.identityFilter)
+	if err != nil {
+		return fmt.Errorf("could get consensus participants at latest finalized block: %w", err)
+	}
+
+	f.mu.Lock()
+	defer f.mu.Unlock()
+
+	if f.lastFinalizedSnapshot != nil && f.lastFinalizedSnapshot.head.Height >= head.Height {
+		return nil
+	}
+
+	f.lastFinalizedSnapshot = &finalizedSnapshot{
+		head:         head,
+		participants: participants,
+	}
+
+	return nil
+}
+
+func (f *FinalizedSnapshotCache) Ready() <-chan struct{} {
+	f.lm.OnStart(func() {
+		go f.finalizationProcessingLoop(f.ctx)
```

Comment:
What is the advantage of using extra channel to track closing + lifecycle manager + manual spawn of goroutine instead of using `unit.Launch` ?